From: Mike Travis <travis@sgi.com>
Subject: x86: Unlock nmi lock after kdb_ipi call
References: bnc#745424
Patch-mainline: not applicable

Fix problem where a lock is left lock after a call to kdb_ipi.

Signed-off-by: Mike Travis <travis@sgi.com>
Acked-by: Jiri Kosina <jkosina@suse.cz>
---
 arch/x86/kernel/traps.c |    1 +
 1 file changed, 1 insertion(+)

--- linux.orig/arch/x86/kernel/traps.c
+++ linux/arch/x86/kernel/traps.c
@@ -439,6 +439,7 @@ static notrace __kprobes void default_do
 	 * to an KDB requested IPI.  If so, kdb will handle it.
 	 */
 	if (kdb_ipi(regs, NULL)) {
+		raw_spin_unlock(&nmi_reason_lock);
 		return;
 	}
 #endif /* defined(CONFIG_SMP) && defined(CONFIG_KDB) */
